presto-spi/src/main/java/io/prestosql/spi/connector/Connector.java

```diff
@@ -155,6 +156,14 @@ default ConnectorAccessControl getAccessControl()
         throw new UnsupportedOperationException();
     }
 
+    /**
+     * @throws UnsupportedOperationException if this connector does not have an EventListener
+     */
+    default EventListener getEventListener()
```

Comment:
We could make this `Iterable<EventListener>` and return `emptyList()` by default, which seems cleaner as we aren't arbitrarily restricting to a single listener, and we don't need to deal with `UnsupportedOperationException`.

Reply:
Fine with me, that's how it looked like in the first approach.